### PR TITLE
 Add cluster to SDS cache key

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

**Review by adiprerepa — COMMENTED** [review]
why are we doing this? I don't have enough context

**Comment by howardjohn** [comment]
Seeing some issues in https://github.com/istio/istio/pull/33620. Basically if the same secret name with different contents is in different cluster we may cache the wrong one

File: pilot/pkg/xds/sds.go [hunk 1]
```diff
@@ -42,10 +42,11 @@ type SecretResource struct {
 	Name         string
 	Namespace    string
 	ResourceName string
+	Cluster      string
 }
 
 func (sr SecretResource) Key() string {
-	return "sds://" + sr.Type + "/" + sr.Name + "/" + sr.Namespace
+	return "sds://" + sr.Type + "/" + sr.Name + "/" + sr.Namespace + "/" + sr.Cluster
```
Comment (adiprerepa) [inline pilot/pkg/xds/sds.go:49]: it doesn't matter, but doesn't it make more sense for the ordering to be:

```
sds://<Cluster>/<Namespace>/<Type>/<Name>
```

**Review by adiprerepa — APPROVED** [review]
lgtm

**Pull request merged**